### PR TITLE
Turret rejection

### PR DESCRIPTION
**Pull request by james-ward**
Reject opposition cargo through turret if possible.

Note that this does not address the auto-retraction issues raised in the discussion of #157

### CODE REVIEW

File: controllers/shooter.py [hunk 1]
```diff
@@ -164,15 +177,14 @@ def firing(self, initial_call) -> None:
...
                 ]
             )
-        if self.flywheels_running:
-            if self.interpolation_override:
-                self.shooter.motor_speed = self.flywheel_speed
-            else:
-                self.shooter.motor_speed = interpolate(
-                    self.distance, self.ranges_lookup, self.flywheel_speed_lookup
-                )
+        self._track()
```
Comment (auscompgeek) [inline controllers/shooter.py:180]: Looks like this will adjust the flywheel speeds if we start to fire whilst we're moving. Is this intended?

Comment (james-ward) [reply controllers/shooter.py:180]: Yes, I think we should have the most up to date range and bearing info for our shots right up to the moment it comes out the turret.

Comment (james-ward) [reply controllers/shooter.py:180]: And if not, those statements were redundant anyway because the stored values were never changed in that state.

Comment (auscompgeek) [reply controllers/shooter.py:180]: Doesn't the shooter speed reset to 0 if it's not commanded each control loop?

**Review by OliverW10 — APPROVED** [review]
This is much cleaner, other than what davo said about the shooter speed resetting I am happy if we merge this and ill create the changes to the auto-retract behaviour on top of it.

**Pull request merged**